### ISSUE TITLE
false `Protocol` incompatibility based on parameter name

### ISSUE THREAD

**Issue by simondoesstuff** [DetachHead/basedpyright#1261]
### Description

[For example,](https://basedpyright.com/?typeCheckingMode=all&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYA0UIArijFgKYD6pAFs6QNYCGAIwrMAUGIAC9Ri3Zdeg4SNIU+AZ1VQAYrggAKImBLkKASgBcIqFagATZsCgRmMDmBu7VzCsBppmKZhAkUlYUPiczTEYTKABaAD4omAtrVKgAOkyxJRV1KAARQKQAN2YbFOs7BycXNw8vHyhVBG4kYGDQ8OZI1BgYhKSKtNpnOhAUJpbSNo6wp2ziviCFbq0dKABeAqLS9xMRNU88TFVUVRg+FFJmXUXloWYabRAIGKgAYigAFXpmIA)
```python
from typing import Protocol, runtime_checkable


@runtime_checkable
class Form(Protocol):
    def method(self, generic_name: int) -> int:
        ...


class Derived:
    def method(self, specific_name: int) -> int:
        return specific_name


variable: Form = Derived()  # False Error
assert isinstance(variable, Form)  # True
```

It complains,
```
Type "Derived" is not assignable to declared type "Form"
  "Derived" is incompatible with protocol "Form"
    "method" is an incompatible type
      Type "(specific_name: int) -> int" is not assignable to type "(generic_name: int) -> int"
        Parameter name mismatch: "generic_name" versus "specific_name"
```

This can occur in multiple scenarios including variable and parameter assignment.
The final line in the example illustrates that the Protocol **is** in fact compatible.

**Comment by KotlinIsland** [comment]
```py
variable.method(generic_name=1) # it's not compatible 
```

`runtime_checkable` shouldn't be used, it's highly unsafe

**Comment by simondoesstuff** [comment]
👍

**Comment by KotlinIsland** [comment]
@DetachHead what do you think of showing a warning when `runtime_checkable` is used?

**Comment by DetachHead** [comment]
you mean because it only checks for method names but not that they have matching signatures? yeah probably, separate issue tho
